### PR TITLE
🐛 Fixup resiliency testing

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
The PR makes various improvements and fixes in resilience testing.

## Related issue(s)

This is intended to causes of flakiness that I know about now. The flakiness is part of #1951 .

### CODE REVIEW

File: .github/workflows/pr-test-e2e.yml [hunk 1]
```diff
@@ -68,7 +68,14 @@ jobs:
...
         if: always()
-        run: kubectl --context kind-kubeflex get pods -A
+        run: |
+          date
+          kubectl --context kind-kubeflex get pods -A
+          kubectl --context kind-kubeflex get pods -A | grep -vw Running | grep -vw Completed | grep -v NAME | while read ns name rest; do echo; kubectl --context kind-kubeflex describe pod -n $ns $name; echo; kubectl --context kind-kubeflex logs -p -n $ns $name || true; echo; kubectl --context kind-kubeflex logs -n $ns $name || true; done
+
+      - name: show Deployment objects in hosting cluster
+        if: always()
+        run: kubectl --context kind-kubeflex get deployments -A
```
Comment (effi-ofer) [inline .github/workflows/pr-test-e2e.yml:78]: Won't the logs of all the pods here make redundant the logs we do in subsequent sections, like ' show kubestellar controller manager log' ?

Comment (MikeSpreitzer) [reply .github/workflows/pr-test-e2e.yml:78]: Right now we have an overlap, which is better than a gap. Neither set covers the other.
We have some logging for every pod that is neither Running nor Completed.
We have some logging for a few pods regardless of their phase.

**Comment by effi-ofer** [comment]
> > util.DeletePods(ctx, coreCluster, "kubeflex-system", "")
> 
> This line all by itself has inherent flakiness: there is no guarantee of the order in which the pods are deleted, and if the postgress pod is deleted first then the other deletions may fail.
> 
> But I do not think that I have observed this failure mode.

Hey @MikeSpreitzer I am going to LGTM this PR, but I have a question about the above comment.  Does the order in which pods are killed makes a difference?  How?  In real life one pod can be killed before another and if this somehow breaks us, then this is something we should resolve?  Or do you mean the test itself is broken somehow?

**Review by effi-ofer — APPROVED** [review]

**Comment by MikeSpreitzer** [comment]
@effi-ofer, regarding my comment about `util.DeletePods(ctx, coreCluster, "kubeflex-system", "")`: The problem was not the mere fact that multiple pods are deleted, nor the fact that the order of deletion is not guaranteed, but the combination of those with not using a retry loop. This PR fixed that.

**Pull request merged**